lib/trie/trie.go

```diff
@@ -390,16 +395,19 @@ func (t *Trie) insertInLeaf(parentLeaf *node.Leaf, key,
 		Generation: t.generation,
 		Dirty:      true,
 	}
+	newBranchParent.AddDescendants(1)
+	nodesCreated++
 
-	return newBranchParent
+	return newBranchParent, nodesCreated
```

Comment:
is `nodesCreated` always 0 or 1 or am i missing something here?

Reply:
It can be 2 if the else block on line 382 is run